Microsoft.Toolkit.Uwp/Helpers/StorageFileHelper.cs

```diff
@@ -588,6 +589,76 @@ public static Task<byte[]> ReadBytesFromLocalFileAsync(string fileName)
+        /// <param name="folder">
+        /// The <see cref="StorageFolder"/> to look for the file in.
+        /// </param>
+        /// <param name="fileName">
+        /// The <see cref="string"/> filename of the file to search for. Must include the file extension and is not case-sensitive.
+        /// </param>
+        /// <param name="isRecursive">
+        /// The <see cref="bool"/>, indicating if the subfolders should also be searched through.
+        /// </param>
+        /// <returns>
+        /// Returns true, if the file exists.
+        /// </returns>
+        public static Task<bool> FileExistsAsync(this StorageFolder folder, string fileName, bool isRecursive = false)
+            => isRecursive
+                ? FileExistsInSubtreeAsync(folder, fileName)
+                : FileExistsInFolderAsync(folder, fileName);
+
+        /// <summary>
+        /// Gets a value indicating whether a file exists in the current folder.
+        /// </summary>
+        /// <param name="folder">
+        /// The <see cref="StorageFolder"/> to look for the file in.
+        /// </param>
+        /// <param name="fileName">
+        /// The <see cref="string"/> filename of the file to search for. Must include the file extension and is not case-sensitive.
+        /// </param>
+        /// <returns>
+        /// Returns true, if the file exists.
+        /// </returns>
+        internal static async Task<bool> FileExistsInFolderAsync(StorageFolder folder, string fileName)
+        {
+            var item = await folder.TryGetItemAsync(fileName);
+            return (item != null) && item.IsOfType(StorageItemTypes.File);
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether a file exists in the current folder or in one of its subfolders.
+        /// </summary>
+        /// <param name="rootFolder">
+        /// The <see cref="StorageFolder"/> to look for the file in.
+        /// </param>
+        /// <param name="fileName">
+        /// The <see cref="string"/> filename of the file to search for. Must include the file extension and is not case-sensitive.
+        /// </param>
+        /// <returns>
+        /// Returns true, if the file exists.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Exception thrown if the <paramref name="fileName"/> contains a quotation mark.
+        /// </exception>
+        internal static async Task<bool> FileExistsInSubtreeAsync(StorageFolder rootFolder, string fileName)
+        {
+            if (fileName.IndexOf('"') >= 0)
+            {
+                throw new ArgumentException(nameof(fileName));
+            }
+
+            var options = new QueryOptions
+            {
+                FolderDepth = FolderDepth.Deep,
+                UserSearchFilter = $"filename:=\"{fileName}\"" // “:=” is the exact-match operator
+            };
+
+            var files = await rootFolder.CreateFileQueryWithOptions(options).GetFilesAsync();
```

Comment:
No reason for thread switching. Should have `.ConfigureAwait(false)` at the end.